x-pack/dev-tools/functional_tests/lib/run_es_with_xpack.js

```diff
@@ -5,46 +5,45 @@
  */
 
 import { resolve } from 'path';
-import { createTestCluster } from '../../../../src/test_utils/es/es_test_cluster';
 import { log } from './log';
+import { Cluster } from '@kbn/es';
+import { RELATIVE_ES_BIN } from './paths';
 
 import { setupUsers, DEFAULT_SUPERUSER_PASS } from './auth';
 
-export async function runEsWithXpack({ ftrConfig, useSAML = false, from }) {
-  const cluster = createTestCluster({
-    port: ftrConfig.get('servers.elasticsearch.port'),
+export async function runEsWithXpack({ procs, config }) {
+  const cluster = new Cluster();
+  const pkg = require('../.././../../package.json');
+  const basePath = resolve(__dirname, '../../../../.es');
+  const randomHash = Math.random().toString(36).substring(2);
+  const clusterName = `test-${randomHash}`;
+  const installOptions = {
+    version: pkg.version,
+    installPath: resolve(basePath, clusterName),
+    sourcePath: resolve(__dirname, '../../../../../elasticsearch'),
     password: DEFAULT_SUPERUSER_PASS,
-    license: 'trial',
-    from,
-    log,
-  });
-
-  const kibanaPort = ftrConfig.get('servers.kibana.port');
-  const idpPath = resolve(
-    __dirname,
-    '../../../test/saml_api_integration/fixtures/idp_metadata.xml'
-  );
+    license: config.get('env').elasticsearch.license,
+    basePath,
+  };
 
-  const esArgs = [
-    'xpack.security.enabled=true',
-  ];
+  // TODO: use installSnapshot by default
+  // use installSource for jenkins CI
+  const { installPath } = config.get('env').from === 'source'
+    ? await cluster.installSource(installOptions)
+    : await cluster.installSnapshot(installOptions);
 
-  const samlEsArgs = [
-    ...esArgs,
-    'xpack.security.authc.token.enabled=true',
-    'xpack.security.authc.token.timeout=15s',
-    'xpack.security.authc.realms.saml1.type=saml',
-    'xpack.security.authc.realms.saml1.order=0',
-    `xpack.security.authc.realms.saml1.idp.metadata.path=${idpPath}`,
-    'xpack.security.authc.realms.saml1.idp.entity_id=http://www.elastic.co',
-    `xpack.security.authc.realms.saml1.sp.entity_id=http://localhost:${kibanaPort}`,
-    `xpack.security.authc.realms.saml1.sp.logout=http://localhost:${kibanaPort}/logout`,
-    `xpack.security.authc.realms.saml1.sp.acs=http://localhost:${kibanaPort}/api/security/v1/saml`,
-    'xpack.security.authc.realms.saml1.attributes.principal=urn:oid:0.0.7',
-  ];
+  const esArgs = config.get('elasticsearchServerArgs');
 
-  await cluster.start(useSAML ? samlEsArgs : esArgs);
-  await setupUsers(log, ftrConfig);
+  await myClusterStart({ procs, esArgs, installPath });
+  //await cluster.start(installPath, esArgs);
+  await setupUsers(log, config);
+}
 
-  return cluster;
+function myClusterStart({ procs, esArgs, installPath }) {
+  return procs.run('es', {
+    cmd: RELATIVE_ES_BIN,
+    args: esArgs.reduce((acc, arg) => acc.concat(['-E', arg]), []),
+    cwd: installPath,
+    wait: /^\[.+?\]\[.+?\]\[.+?\] \[.+?\] started$/,
+  });
```

Comment:
@tylersmalley Your thoughts on the changes here? Did some work last Thursday to use kbn/es here, but still using proc.

Reply:
(I named this `myClusterStart` just to try this out. Name needs to be better)